run_mypy.py

```diff
@@ -93,6 +90,38 @@ def check_mypy() -> None:
         print('mypy >=0.812 is required, older versions report spurious errors')
         sys.exit(1)
 
+class PlatformDict(UserDict):
+
+    def __init__(self, platform: T.Optional[str] = None):
+        super().__init__()
+        self.platform = platform
+
+    def append(self, module: str, platform: T.Optional[str]) -> None:
+        p = self.platform or platform or sys.platform
+        d = self.data.setdefault(p, [])
+        d.append(module)
+
+    def extend(self, modules: T.Dict[str, T.Optional[str]]) -> None:
+        for m, t in modules.items():
+            self.append(m, t)
+
+    def run_mypy(self, command: T.List[str], python_minor_version: T.Optional[int], quiet: bool, root: Path) -> int:
+        retcode = 0
+
+        if python_minor_version is not None and not quiet:
+            print(f'Checking mypy with python version: 3.{python_minor_version}')
+        for platform, modules_to_check in self.data.items():
+            if not quiet:
+                print(f'{platform}:', end=' ')
+            parg = [f'--platform={platform}']
```

Comment:
I haven't checked if it works, but can't be the same be achieved via mypy configuration? Something like
```toml
[mypy-mesonbuild.mtest]
platform = 'linux'
[mypy-mesonbuild.utils.posix]
platform = 'linux'
[mypy-mesonbuild.utils.win32]
platform = 'win32'
```
should work.

```

Reply:
Please disregard. The `platform` option cannot be specified per module.

Reply:
However, the mypy documentation says that you can get the same effect adding an `assert sys.platform == 'win32'`  or `assert sys.platform != 'win32'` at the top of these modules https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks

Reply:
My approach has the advantage to allow forcing a different platform, in case you want to test it.

The problem with the assert is that modules like `mtest` and `minstall` are not platform specific, but use many platform specific functions. My first attempt in fixing them resulted in ugly hacks like in `mlog`. Therefore, it is probably better to simply force the check to be done with linux platform.